### PR TITLE
Add CI_JOB_NAME_SLUG

### LINKED ISSUE #650: Missing predefined variables. E.g. CI_JOB_NAME_SLUG
**Minimal .gitlab-ci.yml illustrating the issue**
```yml
---
syntax shell:bash:
  script:
    - echo "$CI_JOB_NAME_SLUG"
```

Outputs nothing.

**Expected behavior**
Should output: `syntax-shell-bash`

**Host information**
macOS Ventura 13.0
gitlab-ci-local 4.34.0

**Additional context**
CI_JOB_NAME_SLUG is an example. There are several missing predefined variables.

https://docs.gitlab.com/ee/ci/variables/predefined_variables.html.  
Relevant file: src/predefined-variables.ts

### PR DESCRIPTION
**Pull request by firecow**
closes #650

### CODE REVIEW

File: src/job.ts [hunk 1]
```diff
@@ -123,12 +123,13 @@ export class Job {
...
         predefinedVariables["CI_PIPELINE_IID"] = `${this.pipelineIid}`;
         predefinedVariables["CI_JOB_NAME"] = `${this.name}`;
+        predefinedVariables["CI_JOB_NAME_SLUG"] = `${this.name.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").replace(/-$/, "").slice(0, 63).toLowerCase()}`;
         predefinedVariables["CI_JOB_STAGE"] = `${this.stage}`;
         predefinedVariables["CI_PROJECT_DIR"] = ciProjectDir;
         predefinedVariables["CI_JOB_URL"] = `https://${gitData.remote.host}/${gitData.remote.group}/${gitData.remote.project}/-/jobs/${this.jobId}`; // Changes on rerun.
         predefinedVariables["CI_PIPELINE_URL"] = `https://${gitData.remote.host}/${gitData.remote.group}/${gitData.remote.project}/pipelines/${this.pipelineIid}`;
         predefinedVariables["CI_ENVIRONMENT_NAME"] = this.environment?.name ?? "";
-        predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/\/|\s/g, "-").toLowerCase() ?? "";
+        predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").replace(/-$/, "").slice(0, 23).toLowerCase() ?? "";
```
Comment (atli-c) [inline src/job.ts:132]:
```suggestion
        predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").slice(0, 23).replace(/-$/, "").toLowerCase() ?? "";
```

File: src/job.ts [hunk 2]
```diff
@@ -123,12 +123,13 @@ export class Job {
         predefinedVariables["CI_PIPELINE_ID"] = `${this.pipelineIid + 1000}`;
         predefinedVariables["CI_PIPELINE_IID"] = `${this.pipelineIid}`;
         predefinedVariables["CI_JOB_NAME"] = `${this.name}`;
+        predefinedVariables["CI_JOB_NAME_SLUG"] = `${this.name.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").replace(/-$/, "").slice(0, 63).toLowerCase()}`;
```
Comment (atli-c) [inline src/job.ts:126]:
```suggestion
        predefinedVariables["CI_JOB_NAME_SLUG"] = `${this.name.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").slice(0, 63).replace(/-$/, "").toLowerCase()}`;
```

File: src/job.ts [hunk 3]
```diff
@@ -123,12 +123,13 @@ export class Job {
...
         predefinedVariables["CI_JOB_STAGE"] = `${this.stage}`;
         predefinedVariables["CI_PROJECT_DIR"] = ciProjectDir;
         predefinedVariables["CI_JOB_URL"] = `https://${gitData.remote.host}/${gitData.remote.group}/${gitData.remote.project}/-/jobs/${this.jobId}`; // Changes on rerun.
         predefinedVariables["CI_PIPELINE_URL"] = `https://${gitData.remote.host}/${gitData.remote.group}/${gitData.remote.project}/pipelines/${this.pipelineIid}`;
         predefinedVariables["CI_ENVIRONMENT_NAME"] = this.environment?.name ?? "";
-        predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/\/|\s/g, "-").toLowerCase() ?? "";
+        predefinedVariables["CI_ENVIRONMENT_SLUG"] = this.environment?.name?.replace(/[^a-z\d]+/ig, "-").replace(/^-/, "").replace(/-$/, "").slice(0, 23).toLowerCase() ?? "";
         predefinedVariables["CI_ENVIRONMENT_URL"] = this.environment?.url ?? "";
```
Comment (atli-c) [inline src/job.ts:133]:
```suggestion
        predefinedVariables["CI_ENVIRONMENT_URL"] = this.environment?.url ?? "";
        predefinedVariables["CI_ENVIRONMENT_TIER"] = this.environment?.deployment_tier ?? "";
        predefinedVariables["CI_ENVIRONMENT_ACTION"] = this.environment?.action ?? "";
        
```

**Review by atli-c — CHANGES_REQUESTED** [review]
fixes trailing hyphen on slugs. Adds a few more variables

**Comment by atli-c** [comment]
Also:
```sh
CI_COMMIT_AUTHOR="${GITLAB_USER_NAME} <${GITLAB_USER_EMAIL}>"
```

**Comment by atli-c** [comment]
Solves #650

**Comment by firecow** [comment]
@atc-cegodk I cannot merge for some reason.
Even though I've resolved all comments.

**Review by atli-c — APPROVED** [review]
LGTM

**Pull request merged**